### PR TITLE
plugins/lsp/harper-ls: init

### PR DESCRIPTION
**Pull request by Scrumplex**
Harper is a grammar/spell checker "for developers".

https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#harper_ls

https://github.com/elijah-potter/harper

### CODE REVIEW

**Comment by Scrumplex** [comment]
CI seems to be failing as the nixpkgs revision doesn't seem to include https://github.com/NixOS/nixpkgs/pull/335659 yet

**Comment by khaneliman** [comment]
Could you pull the flake.lock commit out? We handle that separately in https://github.com/nix-community/nixvim/pull/2127 when we don't have any major blockers from nixos-unstable. Hopefully next bump will have the main blockers fixed and then we will be able to merge this.

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -212,6 +212,12 @@ let
         "graphql-language-service-cli"
       ];
     }
+    {
+      name = "harper-ls";
+      description = "The Grammar Checker for Developers";
+      serverName = "harper_ls";
+      package = pkgs.harper;
```
Comment (MattSturgeon) [inline plugins/lsp/language-servers/default.nix:219]: This was recently changed, we now expect the form:
```suggestion
      package = "harper";
```

This is evaluated in the form `getAttrFromPath (toList package) pkgs`, and is used to allow showing a user-friendly `defaultText` in the docs, i.e. `"pkgs.${concatStringSep "." (toList package)}"`

**Comment by khaneliman** [comment]
@Scrumplex Alright, flake updated. Now it can be rebased.

**Review by khaneliman — APPROVED** [review]
LGTM, thanks for the addition!

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**